### PR TITLE
Metadata Health Checks

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Create test to ensure SDK metadata RPCs and Explorers are healthy
- Exercise explorer link utils 
- Enable `sort-keys` eslint rule for SDK metadata
- Cleanup metadata and remove unhealthy RPC urls
- Remove NeutronTestnet

### Related issues

Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2997

### Backward compatibility

Yes

### CODE REVIEW

File: typescript/sdk/src/consts/chainMetadata.ts [hunk 1]
```diff
@@ -11,103 +11,116 @@ import { Chains, Mainnets, Testnets } from './chains';
 export const avaxToken = { decimals: 18, name: 'Avalanche', symbol: 'AVAX' };
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chainMetadata.ts:11]: Sorry for the noise here, the eslint sort-keys fixer reordered a bunch of stuff.
The actual changes are: 
1. Remove NeutronTestnet
2. Remove healthy RPC URLs

File: typescript/sdk/src/consts/multisigIsm.ts [hunk 2]
```diff
@@ -2,24 +2,34 @@ import { MultisigConfig } from '../ism/types';
 import { ChainMap } from '../types';
```
Comment (jmrossy) [inline typescript/sdk/src/consts/multisigIsm.ts:2]: Just key reordering

File: typescript/infra/config/environments/testnet4/validators.ts [hunk 3]
```diff
@@ -323,21 +323,21 @@ export const validatorChainConfig = (
...
-            '0xb57486243ce3bb3c38c50a582b8bbd20cb393589',
-            '0x661faee997654d14ead4ae48035883f05c3150cf',
-          ],
-          [Contexts.ReleaseCandidate]: [],
-          [Contexts.Neutron]: [],
-        },
-        'neutrontestnet',
-      ),
-    },
+    // neutrontestnet: {
```
Comment (nambrot) [inline typescript/infra/config/environments/testnet4/validators.ts:326]: IMO let's just remove

File: typescript/sdk/src/consts/multisigIsm.ts [hunk 4]
```diff
@@ -2,24 +2,34 @@ import { MultisigConfig } from '../ism/types';
...
 export const defaultMultisigConfigs: ChainMap<MultisigConfig> = {
-  // ----------------- Mainnets -----------------
-  celo: {
+  alfajores: {
     threshold: 2,
     validators: [
-      '0x63478422679303c3e4fc611b771fa4a707ef7f4a',
-      '0x622e43baf06ad808ca8399360d9a2d9a1a12688b', // dsrv
-      '0xf2c1e3888eb618f1f1a071ef3618f134715a9a49', // everstake
+      '0x2233a5ce12f814bd64c9cdd73410bb8693124d40',
```
Comment (nambrot) [inline typescript/sdk/src/consts/multisigIsm.ts:8]: Can we exclude these objects from the lint rule? Changing the order will I believe cause re-deploy of ISMs which we should avoid (+ i think break the ci fork tests rn)

Comment (nambrot) [reply typescript/sdk/src/consts/multisigIsm.ts:8]: Actually generally seems weird for a lint rule to enforce ordering of string constants in an array?

File: typescript/sdk/src/metadata/health.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      provider.type === ProviderType.CosmJs
+    )
+      resultPromise = isCosmJsProviderHealthy(provider.provider, chainId);
+    else
+      throw new Error(
+        `Unsupported provider type ${provider.type}, new health check required`,
+      );
+    const result = await timeout(
+      resultPromise,
+      5000,
```
Comment (nambrot) [inline typescript/sdk/src/metadata/health.ts:48]: Move out to a constant

File: typescript/sdk/src/test/metadata-check.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    results[metadata.name] = {
+      goodRpcs: 0,
+      badRpcs: 0,
+      goodExplorers: 0,
+      badExplorers: 0,
+    };
+
+    for (const rpc of metadata.rpcUrls) {
```
Comment (nambrot) [inline typescript/sdk/src/test/metadata-check.ts:117]: Should we not parallelize this?

File: typescript/sdk/src/consts/chainMetadata.ts (hunk 1 above)
Comment (nambrot) [reply typescript/sdk/src/consts/chainMetadata.ts:11]: Probably worth a changeset comment for the SDK

File: typescript/sdk/src/consts/multisigIsm.ts (hunk 4 above)
Comment (jmrossy) [reply typescript/sdk/src/consts/multisigIsm.ts:8]: The diff is misleading you here. The eslint fixer didn't reorder any values in the lists here. It reordered the objects at the chain name level.
The fork tests have been broken for a few days at least.

**Comment by jmrossy** [comment]
Confirmation that the env fork test is already broken on main, the ism const changes are aesthetic only.

Screenshot 2023-12-07 at 4 24 53 PM

**Pull request merged**